iib/workers/tasks/utils.py

```diff
@@ -494,3 +623,131 @@ def chmod_recursively(dir_path, dir_mode, file_mode):
             if not os.path.exists(file_path) or os.path.islink(file_path):
                 continue
             os.chmod(file_path, file_mode)
+
+
+def gather_index_image_arches(build_request_config, index_image_infos):
+    """Gather architectures from build_request_config and provided index image.
+
+    :param RequestConfig build_request_config: build request configuration
+    :param dict index_image_infos: dict with index image infos returned
+        by `get_all_index_images_info`
+    :return: set of architecture of all index images
+    :rtype: set
+    """
+    arches = set(build_request_config.add_arches or [])
+    for info in index_image_infos.values():
+        arches |= set(info['arches'])
+
+    if not arches:
+        raise IIBError('No arches were provided to build the index image')
+    return arches
+
+
+def _get_image_arches(pull_spec):
```

Comment:
Could you please rename `_get_image_arches` to `get_image_arches` since it's being imported in other modules.

Reply:
done